app/routes.py

```diff
@@ -1,2 +1,304 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify, make_response, request
+from app import db
+import requests
+import os
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+tasks_bp = Blueprint("tasks", __name__, url_prefix = "/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
 
+
+# THE LINK TO TEST IN HEROKU IS https://task-list-project2.herokuapp.com/tasks
+
+
+@tasks_bp.route("",methods=["POST", "GET"])
+def handle_tasks():
+    
+    # get response body inJSON
+    if request.method == "POST":
+        request_body = request.get_json()
+        # return message if something ismissing
+        if 'title' not in request_body or 'description' not in request_body or 'completed_at' not in request_body:
+            response_body = {
+                'details': 'Invalid data'
+            }
+            return make_response(jsonify(response_body), 400)
+        # making an instance of task after entering info
+        new_task = Task(title=request_body["title"],
+            description=request_body["description"], 
+            completed_at=request_body["completed_at"])
+
+        # add andcommit newtask
+
+        db.session.add(new_task)
+        db.session.commit()
+        # return task to user
+        return {"task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete" : (False
+            if new_task.completed_at == None else True)
+            }},201
+    # get all tasks
+    elif request.method == "GET":
+        title_query = request.args.get("title")
+        if title_query:
+            tasks = Task.query.filter_by(title=title_query)
+        else:
+            tasks = Task.query.all()
+
+
+        sort_query = request.args.get("sort")
+        # account for the asc/desc in wave 2(?)
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()) 
+        else:
+            tasks = Task.query.all()
```

Comment:
We may want to restructure this conditional further. If there is a title AND sort query parameter in the request, then the task table will be queried twice. Once to grab tasks by a specific title then overwritten to grab all tasks ordered by `asc` or `desc`.